templates/common/vsphere/units/nodeip-configuration.service.yaml

```diff
@@ -1,5 +1,5 @@
 name: nodeip-configuration.service
-enabled: true
+enabled: {{ if .Infra -}}{{ if .Infra.Status -}}{{ if .Infra.Status.PlatformStatus -}}{{ if .Infra.Status.PlatformStatus.VSphere -}}{{ if .Infra.Status.PlatformStatus.VSphere.APIServerInternalIP -}}true{{else}}false{{end}}{{end}}{{end}}{{end}}{{end}}
```

Comment:
If I am reading this correctly, this is basically
```
if a
  if b
    if c
      true
    else
      false
    end
  end
end
```

Shouldn't it be `false` for each condition that it doesn't meet?

```
if a
  if b
    if c
      true
    else
      false
    end
  else
    false
  end
else
  false
end
```

such that if any conditions aren't met it still gets written (not empty).

Reply:
@yuqi-zhang  Thanks for the thorough review. I agree with all of this and have pushed the changes. I will need to check CI to make sure the whitespace formatting is good on the new conditional, but I think all is good as far as I know if you want to take another look. 

> I think we should also relax the check in enableUnits function such that if the contents are empty we just no-op (in addition to this PR). Do you think that would work? If so I can try to patch that into 4.8/4.7 so we don't have to block too many edges

This sounds reasonable to me. When you say it will no-op, you mean in the sense of allowing upgrades to progress even if enabled: true contents: <empty>? That sounds like another graceful way of handling this situation we know is problematic.

Reply:
I think the new logic looks correct. Since this is 4.6 only I think I'm ok with the fix as is assuming formatting is correct.

> When you say it will no-op, you mean in the sense of allowing upgrades to progress even if enabled: true contents: ? That sounds like another graceful way of handling this situation we know is problematic.

Yes, I will look to add a check that on empty units (with enable defined) we just output a warning in the MCD logs. We can do it in conjunction with this PR.